packages/snaps-controllers/src/services/offscreen/OffscreenExecutionService.ts

```diff
@@ -35,38 +29,5 @@ export class OffscreenExecutionService extends ProxyExecutionService {
         target: 'child',
       }),
     });
-
-    this.documentUrl = documentUrl;
-  }
-
-  /**
-   * Create a new stream for the specified job. This wraps the runtime stream
-   * in a stream specific to the job.
-   *
-   * @param jobId - The job ID.
-   */
-  protected async initEnvStream(jobId: string) {
-    // Lazily create the offscreen document.
-    await this.#createDocument();
-
-    return super.initEnvStream(jobId);
```

Comment:
Are we guaranteed that it exists by the time this code runs? If not we should be awaiting it somehow

Reply:
Probably not guaranteed, but it also shouldn't matter? `post-message-stream` does a handshake whenever initialised, so the offscreen document will initialise the handshake if loaded after this.

Reply:
It breaks this assumption: https://github.com/MetaMask/snaps/blob/main/packages/snaps-controllers/src/services/proxy/ProxyExecutionService.ts#L76

Reply:
Which probably leads to a race condition

Reply:
Right. 🤔 The offscreen is created asynchronously in the extension at the same time as everything else is loaded. Not sure what the best way to await it is?